src/Features/LanguageServer/Protocol/Features/Options/MetadataAsSourceOptionsStorage.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Options;
+
+namespace Microsoft.CodeAnalysis.MetadataAsSource
+{
+    internal static class MetadataAsSourceOptionsStorage
+    {
+        public static MetadataAsSourceOptions GetMetadataAsSourceOptions(this IGlobalOptionService globalOptions)
+            => new(
+                NavigateToDecompiledSources: globalOptions.GetOption(NavigateToDecompiledSources),
+                AlwaysUseDefaultSymbolServers: globalOptions.GetOption(AlwaysUseDefaultSymbolServers));
+
+        public static Option2<bool> NavigateToDecompiledSources =
+            new("FeatureOnOffOptions", "NavigateToDecompiledSources", defaultValue: true,
```

Comment:
>defaultValue [](http://example.com/codeflow?start=70&length=12)

Use `MetadataAsSourceOptions.Default.NavigateToDecompiledSources` to avoid duplicating the default value